### ISSUE TITLE
[Retail]: RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion is failing

### ISSUE THREAD

**Issue by amanda-tarafa** [GoogleCloudPlatform/dotnet-docs-samples#1937]
Two different errors in diferent runs

```
Failed RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion [3 s]
  Error Message:
   Assert.Contains() Failure
Not found: (filter expression)
In value:  SearchResponse []
  Stack Trace:
     at RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion() in T:\src\github\dotnet-docs-samples\retail\interactive-tutorial\RetailSearch.Samples.Tests\SearchWithQueryExpansionTest.cs:line 34
```

```
Failed RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion [250 ms]
  Error Message:
   System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
  Stack Trace:
     at System.Collections.Generic.List`1.get_Item(Int32 index)
   at RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion() in /tmpfs/src/github/dotnet-docs-samples/retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithQueryExpansionTest.cs:line 31
```

**Comment by chenlei1216** [comment]
This issue has been fixed on the backend and I cannot reproduce it anymore.

**Comment by amanda-tarafa** [comment]
This continues to happen:

On nightly CI, consistently, since October 20.

Just now on presubmit:

Here: https://source.cloud.google.com/results/invocations/3af94d8b-6179-4928-8466-52fb8bec07e5/log
```
Failed RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion [815 ms]
  Error Message:
   System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
  Stack Trace:
     at System.Collections.Generic.List`1.get_Item(Int32 index)
   at RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion() in /tmpfs/src/github/dotnet-docs-samples/retail/interactive-tutorial/RetailSearch.Samples.Tests/SearchWithQueryExpansionTest.cs:line 31
```

Here: https://source.cloud.google.com/results/invocations/348229e4-6c4a-4b7e-81c4-5deaadfb4f04/log

```
Failed RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion [683 ms]
  Error Message:
   System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
  Stack Trace:
     at System.Collections.Generic.List`1.get_Item(Int32 index)
   at RetailSearch.Samples.Tests.SearchWithQueryExpansionTest.TestSearchWithQueryExpansion() in T:\src\github\dotnet-docs-samples\retail\interactive-tutorial\RetailSearch.Samples.Tests\SearchWithQueryExpansionTest.cs:line 31
```

**Comment by chenlei1216** [comment]
Thanks Amanda. It seems that the tests will fail with the current server behavior. Send https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/1965 to fix the test case accordinlgy.

**Comment by amanda-tarafa** [comment]
Thanks! It's green now.

Closed by #1965
